src/main/java/org/apache/mesos/scheduler/SchedulerDriverFactory.java

```diff
@@ -0,0 +1,126 @@
+      final Scheduler scheduler,
+      final FrameworkInfo frameworkInfo,
+      final String masterZkUrl,
+      final byte[] credentialSecret) {
+
+    Credential credential;
+    if (credentialSecret != null && credentialSecret.length > 0) {
+      // User has manually provided a Secret. Provide a Credential with Principal+Secret.
+      if (!frameworkInfo.hasPrincipal() || StringUtils.isEmpty(frameworkInfo.getPrincipal())) {
+        throw new IllegalArgumentException(
+            "Unable to create MesosSchedulerDriver for secret auth, "
+                + "FrameworkInfo lacks required principal: " + frameworkInfo.toString());
+      }
+      credential = Credential.newBuilder()
+          .setPrincipal(frameworkInfo.getPrincipal())
+          .setSecretBytes(ByteString.copyFrom(credentialSecret))
+          .build();
+      LOGGER.info("Creating secret authenticated MesosSchedulerDriver for "
+          + "scheduler = {}, frameworkInfo = {}, masterZkUrl = {}, credentialSecret = {}",
+          scheduler, frameworkInfo, masterZkUrl, credentialSecret);
+    } else if (isSideChannelActive()) {
+      // Sidechannel auth is enabled. Provide a Credential with only the Principal set.
+      if (!frameworkInfo.hasPrincipal() || StringUtils.isEmpty(frameworkInfo.getPrincipal())) {
+        throw new IllegalArgumentException(
+            "Unable to create MesosSchedulerDriver for sidechannel auth, "
+                + "FrameworkInfo lacks required principal: " + frameworkInfo.toString());
+      }
+      credential = Credential.newBuilder()
+          .setPrincipal(frameworkInfo.getPrincipal())
+          .build();
+      LOGGER.info("Creating sidechannel authenticated MesosSchedulerDriver for "
+          + "scheduler = {}, frameworkInfo = {}, masterZkUrl = {}",
+          scheduler, frameworkInfo, masterZkUrl);
+    } else {
+      // No auth. Provide no credential.
+      credential = null;
+      LOGGER.info("Creating unauthenticated MesosSchedulerDriver for "
+          + "scheduler = {}, frameworkInfo = {}, masterZkUrl = {}",
+          scheduler, frameworkInfo, masterZkUrl);
+    }
+
+    return createInternal(scheduler, frameworkInfo, masterZkUrl, credential);
+  }
+
+  /**
+   * Broken out into a separate function to allow testing with custom SchedulerDrivers.
+   */
+  protected SchedulerDriver createInternal(
+      final Scheduler scheduler,
+      final FrameworkInfo frameworkInfo,
+      final String masterZkUrl,
+      final Credential credential) {
+    if (credential == null) {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterZkUrl);
+    } else {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterZkUrl, credential);
+    }
+  }
+
+  /**
+   * Returns whether it appears that sidechannel auth should be used when creating the
+   * SchedulerDriver. Broken out into a separate function to customize behavior in tests.
+   */
+  protected boolean isSideChannelActive() {
```

Comment:
This name is intended to encompass both Bouncer-based Framework Auth as well as potentially other auth in the future which would work via the same mechanism (eg Kerberos). Added a note to the comment explaining this.